### DISCUSSION TITLE
Money cost for commands

### DISCUSSION THREAD

**Discussion by C4BR3R4** [Zrips/CMI#390] [Feature Requests]
**Request** to include in the configuration the possibility of charging money for executing certain commands such as the Essentials plugin: sethome, home, warp, tpa, tpaccept..

_Part of the essentials configuration:_
```
command-costs:
  # /example costs $1000 PER USE
  #example: 1000
  # /kit tools costs $1500 PER USE
  #kit-tools: 1500
set-home: 5000
home: 500
warp: 1000
tpa: 5000
tpaccept: 300
#chat-shout: 500
#chat-question: 500
#msg: 0
#tell: 0
#m: 0
#t: 0
#whisper: 0
#r: 0
#reply: 0
#me: 0
```

I have tried creating CustomAlias, but it does not work correctly, for example, in the tpa/tpaccept.

**Comment by ghost** [comment]
Are you sure this isn't allready possible?

**Comment by C4BR3R4** [comment]
I have not seen any option in the configuration to do it, tell me if I'm wrong.
I have tried several CustomAlias and it does not work correctly.

This is what I have done. I have blocked the original command with the following code:
```
  tpa:
  - msg! [playerName] &c¡No conocemos el conjuro &4/tpa&c! &aPrueba con &2/tpu
```
Then I created the following command:
```
  tpu:
  - perm:quest.mendigo#?!
  - moneycost:5000#?!
  - cmi tpa $1
```
Well, if the player have the money and the permission, it works perfect. But if the player has not the money, the countdown appears, executes tpa command, teleport you to the position, but does not charge you the money.

Sorry bad english.

**Comment by JohOply** [comment]
It would be a really nice addition

**Comment by espen96** [comment]
would absolutely love this addition

**Comment by mrfloris** [comment]
Just like a default cooldown for commands, it would be nice if there's a default cost for commands as well. And that if we use an customalias that this would override it, or be an additional cost.

**Comment by blackfloret** [comment]
This would be very nice
